src/arch/x86_64.rs

```diff
@@ -122,3 +121,22 @@ pub(crate) fn oops(cause: &dyn core::fmt::Display) -> ! {
         }
     }
 }
+
+#[cfg(test)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[repr(u32)]
+pub(crate) enum QemuExitCode {
+    Success = 0x10,
+    Failed = 0x11,
+}
+
+/// Exit using `isa-debug-exit`, for use in tests
+///
+/// NOTE: This is a temporary mechanism until we get proper shutdown implemented
+#[cfg(test)]
+pub(crate) fn qemu_exit(exit_code: QemuExitCode) {
+    let code = exit_code as u32;
+    unsafe {
+        asm!("out 0xf4, eax" :: "{eax}"(code) :: "intel","volatile");
```

Comment:
(oh god does this mean we need to have a style guide)

Reply:
I can use the AT&T syntax, I'm just more used to intel syntax so tend to use it by default.
I don't think we need a style guide because of inline asm, but it's good to try to be generally consistent.

Reply:
i’m fine with either syntax (i actually have a slight preference for intel as well, for extremely silly reasons: `[foo + bar]` is _far_ more obvious to me than...however you say that in at&t, which i actually don’t even remember). 

i think existing inline asm is at&t mostly just because most of it is about one instruction that doesn’t have any memory operands and i was too lazy to type `:::: “intel”`. i would be absolutely fine with changing everything to intel. i’m sure @iximeow has at least one opinion about this...

Reply:
oh another note is that since @iximeow is working on adding disassembly to kernel oopses using `yaxpeax`, and i believe yax’s formatter is intel-ish, i have a slight preference for using the same syntax in the kernel source and in oopses